docs: update DataDog documentation for K8s Daemonset (#291)
convox/convox: docs/integrations/monitoring/datadog.md

You can add operational visibility to your [Rack](../../reference/primitives/rack) with Datadog.

> ## Configure Kubectl to Point at Your Rack

Reviewer: kubectl and kubeconfig should be lowercase and `kubectl` should be backticked anywhere it appears outside of titles

Convox allows you to securely connect your Kubectl to your Convox created Kubernetes cluster by exporting a [Kubeconfig](../../reference/cli/rack#rack-kubeconfig) that will connect you to a Kubernetes API Proxy running inside your Rack. This allows you to use Kubectl without directly exposing the credentials for your Kubernetes cluster. For example if your Rack is called `myrack` you could point your local Kubectl to your Rack cluster as follows

```

> $ convox rack kubeconfig /tmp/myrack-config

Reviewer: Need a `>` in here

> $ export KUBECONTROL=/tmp/myrack-config

Reviewer: Variable is `KUBECONFIG`

```

> This will export the proxy configuration to a temporary file and then point your local Kubecontrol at that location so you can connect to your Rack's cluster. You will need to perform this step before you can execute any Kubectl commands against your cluster.

Reviewer 2: ```suggestion
This will export the proxy configuration to a temporary file and then point your local Kubectl environment at that location so you can connect to your Rack's cluster. You will need to perform this step before you can execute any Kubectl commands against your cluster.
```

> This will export the proxy configuration to a temporary file and then point your local Kubecontrol at that location so you can connect to your Rack's cluster. You will need to perform this step before you can execute any Kubectl commands against your cluster.

Reviewer: `kubectl` not Kubecontrol

## Deploy the Datadog Agent

> Once you have Kubectl pointing at your Rack you can deploy the datadog agent as a Kubernetes Daemonset. The following is based on the [DataDog Documentation](https://docs.datadoghq.com/agent/kubernetes/?tab=daemonset) so please refer back there for any specific tweaks you may want to make.

Reviewer: Datadog should be capitalized properly where it appears. (only first D)

> ## Configure Agent Permissions

Reviewer: I'm wondering if at this point we should just link to the Datadog instructions so we don't have to stay in sync if they change things.

Author: I wanted to do that originally as well but I found the datadog documentation to be a little bit confusing and missing a few important pieces (like getting the base64 encoded key). I like to use myself as reasonable proxy for your average Convox user and I find this step by step to be an improvement.

Reviewer: Does Step 2 in their guide not create the base64-encoded secret?

Reviewer: The Datadog instructions seem to change quite frequently and I worry about these instructions getting out of date. Especially since you are linking to URLs for manifests where the instructions may change.

Reviewer: Just read over their docs more and I can see what you're talking about. They recreate/clobber the secret in their manifest files. That's a shame. Your call. I think we're going to pretty quickly get out of date having our own fork of the instructions but seems like theirs don't work out of the box.

Author: Yeah I struggle with the exact same concerns. I would have loved to just link to their docs but they hung me up and I don't want people to silently struggle. That's why I comprised by including the links to their docs in a few places in case things get out of sync. I vote we merge as is to meet the immediate needs of people who have requested updated instructions.

Reviewer 2: Yeah, I'd say we should merge as well. If Datadog improve, we can adapt. If they just change, but stay broken, we'll have to adapt anyway. 🤷‍♂️

```
$ kubectl get secret datadog-secret -o yaml
apiVersion: v1
data:
  api-key: [BASE64 Encoded Key]
kind: Secret
metadata:
  creationTimestamp: "2020-07-27T20:42:29Z"
  name: datadog-secret
  namespace: default
  resourceVersion: "8898"
  selfLink: /api/v1/namespaces/default/secrets/datadog-secret
  uid: XXXXX-XXXXXX-XXXXX-XXXXXX
type: Opaque

```

## Create a DataDog Agent Manifest

DataDog has several example manifests
- [Manifest with Logs, APM, process, metrics collection enabled.](https://docs.datadoghq.com/resources/yaml/datadog-agent-all-features.yaml)
- [Manifest with Logs, APM, and metrics collection enabled.](https://docs.datadoghq.com/resources/yaml/datadog-agent-logs-apm.yaml)
- [Manifest with Logs and metrics collection enabled.](https://docs.datadoghq.com/resources/yaml/datadog-agent-logs.yaml)
- [Manifest with APM and metrics collection enabled.](https://docs.datadoghq.com/resources/yaml/datadog-agent-apm.yaml)
- [Manifest with Network Performance Monitoring enabled](https://docs.datadoghq.com/resources/yaml/datadog-agent-npm.yaml)
- [Vanilla manifest with just metrics collection enabled.](https://docs.datadoghq.com/resources/yaml/datadog-agent-vanilla.yaml)

> Whichever one these you use as your starting point you will need to put the Base64 encoded key from the previous step in the section at the top where it says

Reviewer 2: ```suggestion
Whichever one of these examples you use as your starting point, you will need to insert the Base64 encoded key from the previous step in the section at the top where it says:
```

```
# Source: datadog/templates/secrets.yaml
# API Key
apiVersion: v1
kind: Secret
metadata:
  name: datadog-agent
  labels: {}
type: Opaque
data:
  api-key: PUT_YOUR_BASE64_ENCODED_API_KEY_HERE
```

> We also recommend adding the following to the environment section of the daeomonset spec:

Reviewer 2: ```suggestion
We also recommend adding the following to the environment section of the Daemonset spec:
```
